test-suite/src/test/scala/org/scalajs/testsuite/javalib/ArraysTest.scala

```diff
@@ -44,22 +44,47 @@ trait ArraysTest extends JasmineTest {
 
   testBody {
 
-    it("should respond to `sort` for Int") {
-      val scalaInts = Array(5, 3, 6, 1, 2, 4)
-      val ints = new Array[Object](scalaInts.length)
-      for (i <- 0 until scalaInts.length)
-        ints(i) = scalaInts(i).asInstanceOf[Object]
-      val sorted = Array(1, 2, 3, 4, 5, 6)
-
-      Arrays.sort(ints, intComparator.asInstanceOf[Comparator[Object]])
-      expect(ints).toEqual(Array(1, 2, 3, 4, 5, 6))
+    def testSort[@specialized T](typeName: String,  elem: Int => T, newArray: Int => Array[T],
```

Comment:
Maybe `@specialized` is needed for top efficiency here, but I would still leave it out (we're in tests).